pkg/controllers/jobset_controller.go

```diff
@@ -524,6 +486,11 @@ func shouldCreateJob(jobName string, ownedJobs *childJobs) bool {
 	return true
 }
 
+func labelAndAnnotateObject(obj metav1.Object, labels map[string]string, annotations map[string]string) {
+	obj.SetLabels(labels)
+	obj.SetAnnotations(annotations)
```

Comment:
does this override or merge with existing maps? we should merge with existing ones.

Reply:
It overrides but I had already cloned the labels/annotations before calling this so it safe to set, however, I refactored this to simpler and do the cloning+merging in one place.